src/snowflake/snowpark/_internal/data_source/utils.py

```diff
@@ -448,7 +458,10 @@ def process_parquet_queue_with_threads(
                 if fetch_with_process:
                     # Check if any processes have failed
                     for i, process in enumerate(workers):
-                        if not process.is_alive() and process.exitcode != 0:
+                        if (
+                            not process.is_alive()
+                            and process.pid not in gracefully_exited_processes
+                        ):
```

Comment:
there is another exitcode usage on line 487, you can do a global search to see if there're any other places